src/flow/domain/mod.rs

```diff
@@ -612,33 +637,44 @@ impl Domain {
                     .expect("migration replay path started with non-materialized node")
                     .lookup(&cols[..], &KeyType::Single(&key[0]));
 
-                let m = if let LookupResult::Some(rs) = rs {
+                if let LookupResult::Some(rs) = rs {
                     use std::iter::FromIterator;
-                    Some(Packet::ReplayPiece {
+                    let m = Some(Packet::ReplayPiece {
                              link: Link::new(source, path[0].0),
                              tag: tag,
-                             context: ReplayPieceContext::Partial { for_key: Vec::from(key) },
+                             context: ReplayPieceContext::Partial { for_key: Vec::from(key), ignore: false },
                              data: Records::from_iter(rs.into_iter().cloned()),
-                         })
+                             transaction_state: transaction_state,
+                    });
+                    (m, source, false)
+                } else if transaction_state.is_some() {
+                    let m = Some(Packet::ReplayPiece {
+                        link: Link::new(source, path[0].0),
+                        tag: tag,
+                        context: ReplayPieceContext::Partial { for_key: Vec::from(key), ignore: true },
+                        data: Vec::<Record>::new().into(),
+                        transaction_state: transaction_state,
+                    });
+                    (m, source, true)
                 } else {
-                    None
-                };
-
-                (m, source)
+                    (None, source, true)
+                }
             }
             _ => unreachable!(),
         };
 
-        if let Some(m) = m {
+        if is_miss {
+            // we have missed in our lookup, so we have a partial replay through a partial replay
+            // trigger a replay to source node, and enqueue this request.
+            self.on_replay_miss(*source.as_local(), Vec::from(key), tag);
+
+            if let Some(m) = m {
+                self.handle_replay(m);
```

Comment:
Also, can we instead move this out of the `is_miss` conditional, and get rid of the `else` clause?